service/history/historyBuilder.go

```diff
@@ -1189,6 +1214,11 @@ func (b *historyBuilder) newStartChildWorkflowExecutionInitiatedEvent(
 		SearchAttributes:             startAttributes.SearchAttributes,
 		ParentClosePolicy:            startAttributes.GetParentClosePolicy(),
 	}
+
+	searchattribute.SetType(
+		attributes.SearchAttributes,
+		searchattribute.ConvertDynamicConfigToIndexedValueTypes(validSearchAttributes()))
+
```

Comment:
start workflow & upsert search attr should be enough
any specific reason to modify start child / continue as new?

Reply:
Child started/continued from old workflow which doesn't have types. Actually, it might not worth it.